src/tests/asyncTests.ts

```diff
@@ -116,28 +117,115 @@ export class AsyncTests {
+          completed: boolean;
+          payload: ApexTestQueueItem;
+        }> => {
+          if (token?.isCancellationRequested) {
+            await this.abortTestRun(testRunId, progress);
+            return {
+              completed: true,
+              payload: {
+                done: true,
+                totalSize: 0,
+                records: []
+              }
+            };
+          }
+
+          progress?.report({
+            type: 'PollingClientProgress',
+            value: 'pollingProcessingTestRun',
+            message: nls.localize('pollingProcessingTestRun', testRunId),
+            testRunId
+          });
+
+          const hasTestSetupTimeField = await this.supportsTestSetupFeature();
+          const testRunSummaryQuery = hasTestSetupTimeField
+            ? `SELECT AsyncApexJobId, Status, ClassesCompleted, ClassesEnqueued, MethodsEnqueued, StartTime, EndTime, TestTime, TestSetupTime, UserId FROM ApexTestRunResult WHERE AsyncApexJobId = '${testRunId}'`
+            : `SELECT AsyncApexJobId, Status, ClassesCompleted, ClassesEnqueued, MethodsEnqueued, StartTime, EndTime, TestTime, UserId FROM ApexTestRunResult WHERE AsyncApexJobId = '${testRunId}'`;
+
+          // Query for test run summary first to check overall status
+          const testRunSummary =
+            await this.connection.tooling.query<ApexTestRunResult>(
+              testRunSummaryQuery
+            );
+
+          if (
+            !testRunSummary ||
+            !testRunSummary.records ||
+            testRunSummary.records.length === 0
+          ) {
+            throw new Error(
+              `No test run summary found for test run ID: ${testRunId}`
+            );
+          }
+
+          const summary = testRunSummary.records[0];
+          const isCompleted = finishedStatuses.includes(summary.Status);
+
+          // Query queue items to get detailed status
+          const queryResult =
+            await this.connection.tooling.query<ApexTestQueueItemRecord>(
+              `SELECT Id, Status, ApexClassId, TestRunResultId, ParentJobId FROM ApexTestQueueItem WHERE ParentJobId = '${testRunId}'`
+            );
+
+          if (!queryResult.records || queryResult.records.length === 0) {
+            throw new Error(
+              `No test queue items found for test run ID: ${testRunId}`
+            );
+          }
+
+          const queueItem: ApexTestQueueItem = {
+            done: isCompleted,
+            totalSize: queryResult.records.length,
+            records: queryResult.records.map((record) => ({
+              Id: record.Id,
+              Status: record.Status,
+              ApexClassId: record.ApexClassId,
+              TestRunResultId: record.TestRunResultId
+            }))
+          };
+
+          return {
+            completed: isCompleted,
+            payload: queueItem
+          };
+        },
+        frequency: POLLING_FREQUENCY,
+        timeout: timeout || POLLING_TIMEOUT
```

Comment:
@CristiCanizales if the caller passes a value of 0 for timeout and the exitOnTestRunId is false the timeout for polling will use the default. 
``` suggestion
        timeout: timeout ?? POLLING_TIMEOUT
```
Not sure what the fallout might be with the suggested change.